src/source/source.js

```diff
@@ -57,6 +57,7 @@ export interface Source {
 
     +onAdd?: (map: Map) => void;
     +onRemove?: (map: Map) => void;
+    +setData?: (data: GeoJSON | string) => Source;
```

Comment:
Leftover -- this can be removed

Reply:
Correct me if I'm wrong, but I believe I still need to add this to the ```Source``` interface so ```setData``` can be called in ```Style#setGeoJSONSourceData```. I'm getting a flow error of ```property setData of unknown type``` when I remove this line.

Reply:
Ah, I see.  `setData` is defined on [GeoJSONSource](https://github.com/mapbox/mapbox-gl-js/blob/master/src/source/geojson_source.js).  So, instead of adding it to the general `Source` interface, you can do the following in `setGeoJSONSourceData`:

```js
const geojsonSource: GeoJSONSource = (this.sourceCaches[id].getSource(): any);
geojsonSource.setData()
```

Note that the use of `: any` essentially disables Flow's checking, so it should be used with caution.  As a general practice, I try to include some sort of runtime check to accompany any use of `: any` -- e.g., `assert(geojsonSource.type === 'geojson')`.

Reply:
:+1: Gotcha -- that makes sense